pkg/reconciler/managed/reconciler.go

```diff
@@ -695,3 +695,16 @@ func (r *Reconciler) Reconcile(req reconcile.Request) (reconcile.Result, error)
 	managed.SetConditions(v1alpha1.ReconcileSuccess())
 	return reconcile.Result{RequeueAfter: r.longWait}, errors.Wrap(r.client.Status().Update(ctx, managed), errUpdateManagedStatus)
 }
+
+// TODO(negz): ReclaimPolicy will be deprecated alongside resource claims; this
+// should be inlined when claims (and thus reclaim policies) are removed.
+func shouldDelete(mg resource.Managed) bool {
+	switch {
+	case mg.GetDeletionPolicy() == v1alpha1.DeletionDelete:
+		return true
+	case mg.GetDeletionPolicy() == v1alpha1.DeletionOrphan:
+		return false
+	}
+
+	return mg.GetReclaimPolicy() == v1alpha1.ReclaimDelete
```

Comment:
Should we rethink the default here? I feel like on this level, the expected behavior is to delete the external resource if I don't say otherwise because that's like lowest level. Also when you delete a `PV`, it doesn't have any policy or whatsoever to manage external deletion; it deletes the external volume immediately. So, deleting by default could feel familiar to people.

Reply:
Whether a PV is deleted or not by default depends on whether or not it was dynamically provisioned, I believe. Orphaning the volume seems like the safest approach, but I agree that given these underlying resources will effectively always be dynamically provisioned (as a product of the composite resource). I could go either way on this one - it could be a little confusing for the deletion policy to have a different default from the reclaim policy during the transition period.

Reply:
One thing to note here is that this would be a behaviour change during the transition period. Currently the default if you omit the reclaim policy is to retain the resource. During the transition if you omitted both policies the default would be to delete the resource.

Reply:
> Whether a PV is deleted or not by default depends on whether or not it was dynamically provisioned, I believe. 

I think we're talking about the EBS volume deletion here rather than `PersistentVolume` deletion here, i.e. the latter policy in PVC<>PV<>external resource. If you manually delete a `PersistentVolume`, it will always delete the external resource in the cloud, AFAIK there is no policy to control that behavior. So, calling `external.Delete` in case of deletion is a must in Kubernetes and we'd be making that behavior default.

> One thing to note here is that this would be a behaviour change during the transition period.

That's true. We're making behavior changes to both policy mechanisms, I guess we have to bite that anyway.